### PR TITLE
Added http endpoint support.

### PR DESCRIPTION
**Pull request by davidlehn**
- Add node API stubs.
- Add http endpoint support.

Work in progress.  HTTP endpoints are doing things direct vs using a node API.

### CODE REVIEW

File: lib/logger.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/*!
+ * Copyright (c) 2016-2018 Digital Bazaar, Inc. All rights reserved.
+ */
+const bedrock = require('bedrock');
+
+module.exports = bedrock.loggers.get('app').child('bedrock-document-server');
```
Comment (mattcollier) [inline lib/logger.js]: convention is not to include `bedrock` in the child logger name.

File: lib/index.js [hunk 2]
```diff
@@ -0,0 +1,323 @@
+/*!
+ * Copyright (c) 2018 Digital Bazaar, Inc. All rights reserved.
+ */
+const async = require('async');
```
Comment (mattcollier) [inline lib/index.js]: strict mode.

File: test/test.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) 2018 Digital Bazaar, Inc. All rights reserved.
+ */
+const bedrock = require('bedrock');
+
+// load plugins
+require('bedrock-server');
+require('bedrock-document-server');
+
+bedrock.events.on('bedrock.init', () => {
```
Comment (mattcollier) [inline test/test.js]: This is not a pattern we've been using.  Is this addressing some difficulty?

File: lib/index.js (hunk 2 above)
Comment (dlongley) [inline lib/index.js]: Don't use `async` in new modules.

File: lib/config.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+require('bedrock-jobs');
+
+// permissions
+const permissions = config.permission.permissions;
+permissions.DOCUMENT_SERVER_ACCESS = {
+  id: 'DOCUMENT_SERVER_ACCESS',
+  label: 'Access Documents',
+  comment: 'Required to access documents.'
+};
+permissions.DOCUMENT_SERVER_CREATE = {
```
Comment (dlongley) [inline lib/config.js]: Should this be `DOCUMENT_CREATE`? Is this for a capability to create a document or a server?

File: test/test.js (hunk 3 above)
Comment (davidlehn) [reply test/test.js]: The difficulty is I threw this together quickly.  It's the same pattern used elsewhere, it just has the jsonld doc loader part pulled out.  Might be able to just remove this.

File: lib/logger.js (hunk 1 above)
Comment (davidlehn) [reply lib/logger.js]: Considering the purpose of namespaces is to add scope clarity, I'm not sure why a convention appeared that uses truncated names.  I guess just to reduce log line length?  Short names increase the possibility of conflicts.

File: lib/index.js (hunk 2 above)
Comment (davidlehn) [reply lib/index.js]: From copy & paste to get things rolling.  Will refactor it away moving forward.

File: lib/logger.js [hunk 5]
```diff
@@ -3,4 +3,4 @@
  */
 const bedrock = require('bedrock');
 
-module.exports = bedrock.loggers.get('app').child('bedrock-document-server');
+module.exports = bedrock.loggers.get('app').child('document-server');
```
Comment (mattcollier) [inline lib/logger.js]: I thought the decision was to leave `bedrock` in and if someone is disturbed enough by it they would write a patch in the logger to filter `bedrock` from all the things.

Comment (davidlehn) [reply lib/logger.js]: Yeah, that was written earlier.  Updated.

File: lib/config.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
+/*!
+ * Copyright (c) 2018 Digital Bazaar, Inc. All rights reserved.
+ */
+const bedrock = require('bedrock');
+const config = bedrock.config;
+require('bedrock-jobs');
```
Comment (mattcollier) [inline lib/config.js]: this belongs in index.js

**Pull request merged**